### PR TITLE
fix: don't include error in response

### LINKED ISSUE #2162: FTL http ingress error handling (response payloads)
A customer brought up the fact that they get a lot of the error dump returned on a HTTP ingress error. Should we just return `500` with an empty body rather than sending back error information?

For example, if schema validation fails on a `ftl.call` it will return `500` AND the schema validation error information.

### PR DESCRIPTION
**Pull request by stuartwdouglas**
Also make sure we log errors related to HTTP processing.

closes #2162

### CODE REVIEW

File: backend/controller/ingress/handler.go [hunk 1]
```diff
@@ -18,6 +18,8 @@ import (
 	"github.com/TBD54566975/ftl/internal/model"
 )
 
+const internalServerError = "Internal Server Error"
```
Comment (stuartwdouglas) [inline backend/controller/ingress/handler.go:21]: An open question is if we want to include this message or just an empty string.

File: backend/controller/ingress/handler.go [hunk 2]
```diff
@@ -35,13 +37,15 @@ func Handle(
...
-		http.Error(w, err.Error(), http.StatusInternalServerError)
+		logger.Errorf(err, "failed to resolve route for %s %s", r.Method, r.URL.Path)
+		http.Error(w, internalServerError, http.StatusInternalServerError)
 		return
 	}
 
 	body, err := BuildRequestBody(route, r, sch)
 	if err != nil {
-		http.Error(w, err.Error(), http.StatusBadRequest)
+		// Don't log on bad request, this can be trivially triggered by malicious clients
```
Comment (alecthomas) [inline backend/controller/ingress/handler.go:47]: I think it would be worth logging, but at Debug/Trace, so that it's useful during local development.

**Review by alecthomas — APPROVED** [review]
Noice

File: backend/controller/ingress/handler.go (hunk 2 above)
Comment (stuartwdouglas) [reply backend/controller/ingress/handler.go]: I have added it at debug

File: backend/controller/ingress/handler.go (hunk 1 above)
Comment (alecthomas) [reply backend/controller/ingress/handler.go:21]:
```suggestion
const internalServerError = http.StatusText(http.StatusInternalServerError)
```

(or inline that call elsewhere)

Comment (stuartwdouglas) [reply backend/controller/ingress/handler.go:21]: I don't think that compiles, I would need to inline it

File: backend/controller/ingress/handler.go [hunk 3]
```diff
@@ -35,13 +37,16 @@ func Handle(
...
+		http.Error(w, internalServerError, http.StatusInternalServerError)
 		return
 	}
 
 	body, err := BuildRequestBody(route, r, sch)
 	if err != nil {
-		http.Error(w, err.Error(), http.StatusBadRequest)
+		// Only log at debug, as this is a client side error
+		logger.Debugf("bad request: %s", err.Error())
+		http.Error(w, internalServerError, http.StatusBadRequest)
```
Comment (alecthomas) [inline backend/controller/ingress/handler.go:49]: Actually, this is the error case where we will want to return some subset of the underlying errors. The validation errors returned are intended for the caller.

File: backend/controller/ingress/handler.go (hunk 1 above)
Comment (stuartwdouglas) [reply backend/controller/ingress/handler.go:21]: Although it did make me realize I am sending the wrong message in some places

File: backend/controller/ingress/handler.go (hunk 3 above)
Comment (stuartwdouglas) [reply backend/controller/ingress/handler.go]: good point

File: backend/controller/ingress/handler.go (hunk 1 above)
Comment (alecthomas) [reply backend/controller/ingress/handler.go]: Make it `var ..` to compile.

Comment (stuartwdouglas) [reply backend/controller/ingress/handler.go]: I inlined it as some of the messages were for different response codes

**Pull request merged**